p2p/node.go

```diff
@@ -99,25 +96,27 @@ type Config struct {
 	UseBootstrapList bool
 }
 
-// New creates a new Node with the given config.
-func New(config Config) (*Node, error) {
-
-	nodeCtx, cancel := context.WithCancel(context.Background())
-
+// New creates a new Node with the given context and config. The Node will stop
+// all background operations if the context is canceled.
+func New(ctx context.Context, config Config) (*Node, error) {
```

Comment:
I tried to write the `p2p` package such that `New` does not require a context and doesn't do any processing in the background. Unfortunately it is not practical to do so because of how `libp2p.Host` works. In order to create a `Host`, with `libp2p.New`, you need to pass in a context. So `p2p.New` expects a context and `p2p.Start` re-uses the context provided to `New`.